Ladybird/AppKit/Application/ApplicationDelegate.mm

```diff
@@ -134,6 +149,54 @@ - (void)openLocation:(id)sender
     [controller focusLocationToolbarItem];
 }
 
+- (void)setAutoPreferredColorScheme:(id)sender
+{
+    m_preferred_color_scheme = Web::CSS::PreferredColorScheme::Auto;
+    [self broadcastPreferredColorSchemeUpdate];
+}
+
+- (void)setDarkPreferredColorScheme:(id)sender
+{
+    m_preferred_color_scheme = Web::CSS::PreferredColorScheme::Dark;
+    [self broadcastPreferredColorSchemeUpdate];
+}
+
+- (void)setLightPreferredColorScheme:(id)sender
+{
+    m_preferred_color_scheme = Web::CSS::PreferredColorScheme::Light;
+    [self broadcastPreferredColorSchemeUpdate];
+}
+
+- (void)broadcastPreferredColorSchemeUpdate
+{
+    auto* auto_scheme_item = [self.color_scheme_menu itemWithTag:COLOR_SCHEME_MENU_AUTO_TAG];
+    auto* dark_scheme_item = [self.color_scheme_menu itemWithTag:COLOR_SCHEME_MENU_DARK_TAG];
+    auto* light_scheme_item = [self.color_scheme_menu itemWithTag:COLOR_SCHEME_MENU_LIGHT_TAG];
+
+    switch (m_preferred_color_scheme) {
+    case Web::CSS::PreferredColorScheme::Auto:
+        [auto_scheme_item setState:NSControlStateValueOn];
+        [dark_scheme_item setState:NSControlStateValueOff];
+        [light_scheme_item setState:NSControlStateValueOff];
+        break;
+    case Web::CSS::PreferredColorScheme::Dark:
+        [auto_scheme_item setState:NSControlStateValueOff];
+        [dark_scheme_item setState:NSControlStateValueOn];
+        [light_scheme_item setState:NSControlStateValueOff];
+        break;
+    case Web::CSS::PreferredColorScheme::Light:
+        [auto_scheme_item setState:NSControlStateValueOff];
+        [dark_scheme_item setState:NSControlStateValueOff];
+        [light_scheme_item setState:NSControlStateValueOn];
```

Comment:
Is there a way to tie these options into a mutually exclusive group in AppKit? This feels kinda sad to do it manually

Reply:
Not quite, but after some more googling, I found the "standard" way to do this is to override `validateMenuItem:(NSMenuItem*)item` and just check/uncheck the menu items there as they come in. So we don't need to track the menu or its items with tags.